### ISSUE TITLE
Error when clicking on top-left button

### ISSUE THREAD

**Issue by guiled** [FokkeZB/nl.fokkezb.drawer#83]
There is a bug with SDK 6.2.2.GA (and 6.3) on Android when clicking on the hamburger icon to open the left window, with Genymotion Nexus 6.0, running the test project.

Here is the log

```
[WARN]  W/System.err: android.view.ViewRootImpl$CalledFromWrongThreadException: Only the original thread that created a view hierarchy can touch its views.
[WARN]  W/System.err: 	at android.view.ViewRootImpl.checkThread(ViewRootImpl.java:6556)
[WARN]  W/System.err: 	at android.view.ViewRootImpl.invalidateChildInParent(ViewRootImpl.java:942)
[WARN]  W/System.err: 	at android.view.ViewGroup.invalidateChild(ViewGroup.java:5081)
[WARN]  W/System.err: 	at android.view.View.invalidateInternal(View.java:12713)
[WARN]  W/System.err: 	at android.view.View.invalidate(View.java:12677)
[WARN]  W/System.err: 	at android.view.View.invalidate(View.java:12661)
[WARN]  W/System.err: 	at android.support.v4.widget.DrawerLayout.openDrawer(DrawerLayout.java:1615)
[WARN]  W/System.err: 	at android.support.v4.widget.DrawerLayout.openDrawer(DrawerLayout.java:1641)
[WARN]  W/System.err: 	at android.support.v4.widget.DrawerLayout.openDrawer(DrawerLayout.java:1625)
[WARN]  W/System.err: 	at ti.modules.titanium.ui.widget.TiUIDrawerLayout.openLeft(TiUIDrawerLayout.java:193)
[WARN]  W/System.err: 	at ti.modules.titanium.ui.widget.TiUIDrawerLayout.toggleLeft(TiUIDrawerLayout.java:188)
[WARN]  W/System.err: 	at ti.modules.titanium.ui.android.DrawerLayoutProxy.toggleLeft(DrawerLayoutProxy.java:49)
[WARN]  W/System.err: 	at org.appcelerator.kroll.runtime.v8.V8Function.nativeInvoke(Native Method)
[WARN]  W/System.err: 	at org.appcelerator.kroll.runtime.v8.V8Function.callSync(V8Function.java:57)
[WARN]  W/System.err: 	at org.appcelerator.kroll.runtime.v8.V8Function.handleMessage(V8Function.java:82)
[WARN]  W/System.err: 	at android.os.Handler.dispatchMessage(Handler.java:98)
[WARN]  W/System.err: 	at android.os.Looper.loop(Looper.java:148)
[WARN]  W/System.err: 	at org.appcelerator.kroll.KrollRuntime$KrollRuntimeThread.run(KrollRuntime.java:118)
[ERROR] TiExceptionHandler: (main) [1679,7620] ----- Titanium Javascript Runtime Error -----
[ERROR] TiExceptionHandler: (main) [0,7620] - In /alloy/widgets/nl.fokkezb.drawer/controllers/widget.js:280,27
[ERROR] TiExceptionHandler: (main) [0,7620] - Message: Uncaught Only the original thread that created a view hierarchy can touch its views.
[ERROR] TiExceptionHandler: (main) [0,7620] - Source:         return $.instance.toggleLeft();
[ERROR] V8Exception: Exception occurred at /alloy/widgets/nl.fokkezb.drawer/controllers/widget.js:280: Uncaught Only the original thread that created a view hierarchy can touch its views.
[ERROR] V8Exception: Only the original thread that created a view hierarchy can touch its views.
```

**Comment by FokkeZB** [comment]
Thanks. I'm not actively maintaining this project anymore. PRs are welcome and perhaps if you provide some more context (a project - skeleton) that reproduces others could help out.

**Comment by guiled** [comment]
I took the "test" branch, with the widget corresponding to current master branch, so this is the more classical way.
I could make a PR but I'm looking for some informations to help
thank you for your fast support !

**Comment by guiled** [comment]
I may found a way to fix it but I don't know if it doesn't make more bugs elsewhere...
In tiapp.xml I added `<property name="run-on-main-thread" type="bool">true</property>` And there is no problem like this... but is it safe with 6.2 ? 6.3 ?

**Comment by FokkeZB** [comment]
I have no idea. @hansemannn wasn't main thread the default yet?

**Comment by hansemannn** [comment]
Yes. If you don't set it, it will be used by default in 6.x+, at least for iOS. As this is Android, the module should simply dispatch the UI-action mentioned above on the correct thread.

**Comment by guiled** [comment]
Actually this module already uses the internal drawer module of Titanium

https://github.com/FokkeZB/nl.fokkezb.drawer/blob/master/nl.fokkezb.drawer/controllers/widget.js#L109

My previous experience with run-on-main-thread was that it caused weird bugs... But they should be fixed by now, shouldn't they ?

**Comment by FokkeZB** [comment]
@guiled that was a reason change and I don't know the current version on gitt.io doesn't have that. I'll publish a new version with that.

**Comment by hansemannn** [comment]
It's probably [this PR](https://github.com/appcelerator/titanium_mobile/pull/9516) that is fixed for 7.0.0 already.
